backend/x86_ast.mli

```diff
@@ -374,7 +374,7 @@ type asm_line =
   | Byte of constant
   | Bytes of string
   | Comment of string
-  | Global of string
+  | Global of { protected: bool; sym: string }
```

Comment:
It would be better to have `Protected` directive as a separate varient, not part of `Global`, similarly to the way we have `Hidden` separately.